src/AvlTree/AvlTreeNode.ts

```diff
@@ -248,8 +248,31 @@ class AvlTreeNode<ValueType> {
     return new Array<number>();
   }
 
-  public findRange(beginValue: ValueType, endValue: ValueType): number[] | null {
-    return new Array<number>();
+  public findRange(beginValue: ValueType, endValue: ValueType): AvlTreeNode<ValueType>[] | null {
+    let selectedValues = new Array<AvlTreeNode<ValueType>>();
+
+    return this.searchRange(selectedValues, beginValue, endValue);
+  }
+
+  public searchRange(selectedValue: AvlTreeNode<ValueType>[], beginValue: ValueType, endValue: ValueType) : AvlTreeNode<ValueType>[] | null {
```

Comment:
I didn't quite undestand this method logic, but I'll review it later.

Reply:
I have updated this method, please check the newest version.